tests/filtered_relation/tests.py

```diff
@@ -279,28 +287,153 @@ def test_as_subquery(self):
+                book_edited_by_b=FilteredRelation('book', condition=Q(book__editor__name__icontains='b')),
+            ).filter(book_edited_by_b__isnull=False).distinct()
+            list(qs)
 
-    def test_with_foreign_key_on_condition_error(self):
+    def test_with_foreign_key_relation_name_lookup(self):
         msg = (
-            "FilteredRelation's condition doesn't support nested relations "
-            "(got 'book__editor__name__icontains')."
+            "FilteredRelation's relation_name doesn't support lookups "
+            "(got 'book__title__icontains')."
         )
         with self.assertRaisesMessage(ValueError, msg):
-            list(Author.objects.annotate(
-                book_edited_by_b=FilteredRelation('book', condition=Q(book__editor__name__icontains='b')),
-            ))
+            qs = Author.objects.annotate(
+                book_edited_by_b=FilteredRelation(
+                    'book__title__icontains',
+                    condition=Q(book__editor__name__icontains='b')
+                ),
+            ).filter(book_edited_by_b__isnull=False).distinct()
+            list(qs)
+
+    def test_with_nested_filtered_relationship(self):
+        qs = Book.objects.annotate(
+            author_favorite_books=FilteredRelation(
+                'author__favorite_books',
+                condition=Q(author__favorite_books__title__icontains='book')
+            )
+        )
+        qs = qs.values(
+            'title', 'author__name', 'author_favorite_books__title'
+        ).order_by(
+            'title', 'author__name', 'author_favorite_books__title'
+        )
+
+        self.assertSequenceEqual(qs, [
+            {
+                'title': self.book1.title, 'author__name': self.author1.name,
+                'author_favorite_books__title': self.book2.title,
+            },
+            {
+                'title': self.book1.title, 'author__name': self.author1.name,
+                'author_favorite_books__title': self.book3.title,
+            },
+            {
+                'title': self.book4.title, 'author__name': self.author1.name,
+                'author_favorite_books__title': self.book2.title,
+            },
+            {
+                'title': self.book4.title, 'author__name': self.author1.name,
+                'author_favorite_books__title': self.book3.title,
+            },
+            {
+                'title': self.book2.title, 'author__name': self.author2.name,
+                'author_favorite_books__title': None,
+            },
+            {
+                'title': self.book3.title, 'author__name': self.author2.name,
+                'author_favorite_books__title': None,
+            },
+        ])
+
+    def test_with_composed_relations(self):
+        qs = Author.objects.annotate(
+            my_books=FilteredRelation(
+                'book', condition=Q(book__title__icontains='book')
+            )
+        )
+        qs = qs.annotate(
```

Comment:
I would prefer to chain `annotate()`, e.g.
```python
...
).annotate(
...
```

Reply:
addressed!